grease-cli/src/Grease/Main.hs

```diff
@@ -1102,9 +1106,11 @@ simulateLlvmCfg la simOpts bak fm halloc llvmCtx initMem setupHook mbStartupOvCf
     execAndRefine bak (simSolver simOpts) fm la setupAnns initMem heuristics argNames argShapes args bbMapRef (simLoopBound simOpts) execFeats st
 
   res <- case result of
-    RefinementBug b _ -> do
-      let argsJson = []  -- TODO
-      pure (BatchBug (MkBatchBug b argsJson))
+    RefinementBug b cData -> do
+      let argsJson = concArgsToJson fm argNames (concArgs cData) argTys
+      let interestingShapes = interestingConcretizedShapes argNames (initArgs ^. Shape.argShapes) (concArgs cData)
+      let prettyArgs = printConcArgs MM.Addr64 argNames interestingShapes (concArgs cData)
```

Comment:
Now that we are calling `printConcArgs` from both the LLVM and Macaw frontends, it is a little strange that we have to pass `Addr64` (a `macaw`-specific value) in an LLVM-specific context. I propose that we change `printConcArgs` to take a `NatRepr` (or some other data type not from `macaw`) instead of an `AddrWidthRepr`.

Reply:
The difficulty here is that this value gets passed to the shape DSL printer, which only supports specifically 32- or 64-bit addresses. I agree it's a bit strange, but `AddrWidthRepr` is a convenient type for expressing this constraint. Do you think it's worth making our own version of it?

Reply:
Fair enough. Given that we're already depending on `macaw` anyway, I suppose it's not the worst thing in the world to use `AddrWidthRepr` for this purpose. I retract my suggestion.